File: baseutils/src/main/java/com/ec/www/view/swiperecycler/base/multitype/CompareItem.java
package com.ec.www.view.swiperecycler.base.multitype;

/**
 * Created by huang on 2016/12/27.
 */

public interface CompareItem<T, E> {

    boolean areItemsTheSame(T oldItem,
                            E newItem);

    boolean areContentsTheSame(T oldItem,
                               E newItem);

    Object getChangePayload(T oldItem,
                            E newItem);
}

File: baseutils/src/main/java/com/ec/www/view/swiperecycler/base/multitype/CompareItem.kt
package com.ec.www.view.swiperecycler.base.multitype

/**
 * Created by huang on 2016/12/27.
 */

interface CompareItem<T, E> {

    fun areItemsTheSame(oldItem: T,
                        newItem: E): Boolean

    fun areContentsTheSame(oldItem: T,
                           newItem: E): Boolean

    fun getChangePayload(oldItem: T,
                         newItem: E): Any?
}
